pkg/controller/seed-controller-manager/ipam/helper.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2022 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ipam
+
+import (
+	"fmt"
+	"math/big"
+	"net"
+)
+
+var (
+	errIncompatiblePool = fmt.Errorf("pool is incompatible with a current cluster allocation")
+)
+
+type datacenterIPAMPoolUsageMap map[string]struct{}
+
+func newDatacenterIPAMPoolUsageMap() datacenterIPAMPoolUsageMap {
+	return make(datacenterIPAMPoolUsageMap)
+}
+
+func (m datacenterIPAMPoolUsageMap) setUsed(value string) {
+	m[value] = struct{}{}
+}
+
+func (m datacenterIPAMPoolUsageMap) isUsed(value string) bool {
+	_, isUsed := m[value]
+	return isUsed
+}
```

Comment:
Is this just a reimplementation of [`sets.String`](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets#String) ?

Reply:
You are correct. I replaced it with `sets.String`